trinity/main.py

```diff
@@ -137,80 +134,66 @@ def main() -> None:
         log_level
     )
 
-    display_launch_logs(chain_config)
-
-    # if console command, run the trinity CLI
-    if args.subcommand == 'attach':
-        run_console(chain_config, not args.vanilla_shell)
-        sys.exit(0)
-
-    # start the listener thread to handle logs produced by other processes in
-    # the local logger.
-    listener.start()
-
-    extra_kwargs = {
-        'log_queue': log_queue,
-        'log_level': log_level,
-        'profile': args.profile,
-    }
-
-    # First initialize the database process.
-    database_server_process = ctx.Process(
-        target=run_database_process,
-        args=(
-            chain_config,
-            LevelDB,
-        ),
-        kwargs=extra_kwargs,
-    )
+    if hasattr(args, 'func'):
+        args.func(args, chain_config)
+    else:
+        display_launch_logs(chain_config)
```

Comment:
The `main()` method is pretty long. Maybe this `else` block belongs in a new `run_node()` method. (This has the side benefit of not requiring the huge indentation change, especially if you insert it right after `main()`)

Reply:
Yeah, I had that initially. But then, having to pass 6 or 7 arguments to that function turned me off and made me roll it back. Another idea to get rid of the `else` and the indentation would be to do.

```python
if hasattr(args, 'func'):
    return args.func(args, chain_config)
```

Reply:
Ok, moved some code around and now we are down to five parameters which I think is still ok.